how-to/taskcluster/taskcluster_cli.rst

```diff
@@ -59,26 +59,71 @@ avoid a) leaking scopes that can cause a lot of damage, and/or b)
 accidentally running the wrong command in a shell and doing damage to
 the production cluster.
 
-Aliases
-~~~~~~~
-
-By setting aliases in our shells, we can perform common tasks without
-having to memorize the syntax every time. Here are a few aliases that
-may be helpful:
-
-::
-
-   # To set your root URL to the production firefoxci or stage cluster
-   tc-fxci='export TASKCLUSTER_ROOT_URL=https://firefox-ci-tc.services.mozilla.com/'
-   tc-staging='export TASKCLUSTER_ROOT_URL=https://stage.taskcluster.nonprod.cloudops.mozgcp.net'
-
-
-   # To log out explicitly
-   tc-logout='unset TASKCLUSTER_CLIENT_ID; unset TASKCLUSTER_ACCESS_TOKEN'
-
-   # Rerunning and cancelling tasks are a common request/need in releaseduty;
-   # grant this set of scopes for 1 hour
-   tc-relduty=$'eval $(taskcluster signin --expires 1h -s "queue:rerun-task:*\nqueue:cancel-task:*")'
-
-   # Up to root privs: only grant these for 15min
-   tc-signin='eval $(taskcluster signin --expires 15m "$@")'
+Signin Helper
+~~~~~~~~~~~~~
+
+Using this shell helper function is an easy way to ensure only the minimum
+amount of scopes are being used:
+
+.. code-block:: shell
+
+   tc-signin () {
+       if [[ "$#" == "0" ]]; then
+           echo "error: must provide use case"
+           return 1
+       fi
+       purpose="$1"
+       shift
+   
+       case $purpose in
+           ciadmin)
+               expiry="60m"
+               scopes=(
+                   "hooks:list-hooks:*"
+               )
+               ;;
+           relduty)
+               expiry="60m"
+               scopes=(
+                   "queue:rerun-task:*"
+                   "queue:cancel-task:*"
+               )
+               ;;
+           hook)
+               expiry="15m"
+               scopes=(
+                   "hooks:trigger-hook:*"
+               )
+               ;;
+           root)
+               expiry="15m"
```

Comment:
I suppose you could admonish here...although that might get old fast for folks who are using it appropriately - just a thought.

Reply:
Yeah, I'm on the fence there.. I think as long as this isn't the default behaviour (like it is without this function), it gets us what we want.